tests/parsenum/test_parsenum_improper_usage.sh

```diff
@@ -0,0 +1,15 @@
+#!/bin/sh -e
+
+# Check assertion failures, separating lines so that we automatically check
+# that test_parsenum is giving a non-zero exit code.
+output=$(! ./test_parsenum 1 2>&1)
+echo "$output" | grep -q "PARSENUM applied to signed integer without specified bounds"
```

Comment:
Is there a reason you don't want to use a temporary file here?  I'd say that
```
! ./test_parsenum 1 2>tmpfile
grep -q "..." tmpfile
```
is much cleaner than using `$()` and `echo`.